src/net/parse.cpp

```diff
@@ -53,6 +53,21 @@ namespace net
                 return make_error_code(net::error::invalid_port);
         }
 
+        const boost::string_ref::size_type slash = address.find_first_of('/');
+        if (slash != boost::string_ref::npos)
+        {
+          std::uint32_t ip = 0;
+          if (epee::string_tools::get_ip_int32_from_string(ip, std::string{address.data(), slash}))
+          {
+            uint32_t mask;
+            if (!epee::string_tools::get_xtype_from_string(mask, std::string{address.substr(slash + 1)}))
+                return make_error_code(net::error::invalid_mask);
+            if (mask > 32)
+                return make_error_code(net::error::invalid_mask);
+            return {epee::net_utils::ipv4_network_subnet{ip, (uint8_t)mask}};
```

Comment:
This is a re-hash of an old comment but this is now potentially returning subnets where they cannot be used. Its partially due to the hacking of `network_address` for something that isn't a network address. I haven't gone through some of the code paths, but Im not sure what happens if someone passes a `/24`, etc., into the "exclusive node" options, etc. I think this returns an error on all such paths (hopefully).

Reply:
Yes, good point. I did not think about this.